packages/edit-post/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -7,37 +7,63 @@ import { get } from 'lodash';
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
-import { Button } from '@wordpress/components';
+import { Button, Icon } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
 import { wordpress } from '@wordpress/icons';
 
 function FullscreenModeClose() {
-	const { isActive, postType } = useSelect( ( select ) => {
-		const { getCurrentPostType } = select( 'core/editor' );
-		const { isFeatureActive } = select( 'core/edit-post' );
-		const { getPostType } = select( 'core' );
+	const { isActive, isRequestingSiteIcon, postType, siteIconUrl } = useSelect(
+		( select ) => {
+			const { getCurrentPostType } = select( 'core/editor' );
+			const { isFeatureActive } = select( 'core/edit-post' );
+			const { isResolving } = select( 'core/data' );
+			const { getEntityRecord, getPostType } = select( 'core' );
+			const siteData = getEntityRecord( 'root', 'base', undefined ) || {};
 
-		return {
-			isActive: isFeatureActive( 'fullscreenMode' ),
-			postType: getPostType( getCurrentPostType() ),
-		};
-	}, [] );
+			return {
```

Comment:
@mtias @youknowriad do you think we should limit this to just edit-site for now, or do you think it would also be good to start testing it in edit-post?

Reply:
We should do it in both, but if we are not certain yet, better to keep it in edit-site. We would need to decide if it's ready for 5.5 as well (in edit-post).

Reply:
> We would need to decide if it's ready for 5.5 as well

Sounds good 👍 . I'm newer on the team so I'm not sure what this entails. Could you elaborate?

Also, I wanted to call out that I'm on the gardening rotation so development on this PR is going to slow down until the end of the month.

Reply:
The second week of July is the beta period for WordPress 5.5. Anything that affects the post-editor needs to be in good shape by then.

Reply:
@mtias I'm back from the gardening rotation, which leaves us 2 weeks to test this feature. 

Would you be able to send me documentation about how I'd make sure the post-editor is "in good shape" with these changes? Or is this something I'd coordinate with @sirreal?

Reply:
@mtias @vindl Since it sounds like there is more extensive testing needed before releasing this feature in `edit-post`, I'm leaning towards separating this feature into two PRs. The current PR would be for `edit-site`, and the follow-up would be for `edit-post`. 

Since the work here isn't urgent, we wouldn't have to rush testing while we prep for 5.6. Let me know what you think.